File: java-code/src/main/java/refactoring/conditional_logic/special_case/before/Client1.java
package refactoring.conditional_logic.special_case.before;

public class Client1 {
  //Client has to deal with null customer
  public String getName(Site site) {
    var customer = site.getCustomer();

    String customerName;
    if(customer == null) {
      customerName = "occupant";
    } else {
      customerName = customer.getName();
    }

    return customerName;
  }
}

File: go-code/conditional_logic/special_case/before/client1.go
package before

type Client1 struct {
	// Define the Client1 struct if needed
}

func (client *Client1) getName(site Site) string {
	customer := site.Customer

	var customerName string
	if customer == nil {
		customerName = "occupant"
	} else {
		customerName = customer.Name
	}

	return customerName
}
